src/main/java/uk/gov/justice/hmpps/offenderevents/services/EventRetrievalService.java

```diff
@@ -79,4 +84,59 @@ public void pollEvents(final LocalDateTime currentTime) {
             log.warn("Skipping Event Retrieval as start after end, start = {}, end = {}", startTime, endTime);
         }
     }
+
+    /**
+     * For testing run a poll for the immediately ended minute, repeat 1 min later and compare:
+     * <ol>
+     *  <li>run for the latest minute (POLL_NAME_TEST.time -> now - 1 second) remembering time range and count, A
+     *  <li>run for [PREVIOUS_POLL_NAME_TEST -> POLL_NAME_TEST] time range remembering time range and count, B
+     *  <li>compare count B with POLL_NAME_TEST.count - should be the same
+     *  <li>store count A in POLL_NAME_TEST.count
+     *  <li>move POLL_NAME_TEST.time to PREVIOUS_POLL_NAME_TEST.time
+     *  <li>store A end time in POLL_NAME_TEST.time
+     * </ol>
+     */
+    public void runTestPolls(LocalDateTime now) {
+        final LocalDateTime endTimeA = now.minus(1, SECONDS);
+
+        repository.findById(POLL_NAME_TEST).ifPresentOrElse(
+                test -> {
+                    val previousTest = repository.findById(PREVIOUS_POLL_NAME_TEST).orElse(PollAudit.builder()
+                            .pollName(PREVIOUS_POLL_NAME_TEST)
+                            .build());
+
+                    final var eventsA = externalApiService.getEvents(test.getNextStartTime(), endTimeA);
+                    final var countA = eventsA.size();
+                    log.debug("runTestPolls(): A interval {} to {}, count {}", test.getNextStartTime(), endTimeA, countA);
```

Comment:
sorry, struggling to understand what `A` and `B` are here and what is going on.
`A` gets initialised as `now` minus 1 minute and 1 second and `endTimeA` is `now` minus 1 second.  So `A` is the previous minute and `B` is minute before?  Will `countA` here be the same as `test.getNumberOfRecords()`, but you don't seem to be comparing those values?

Reply:
Yes thats right, but im checking that the current B count is the same as the A count of the last poll

Reply:
think it might be clearer for `A` to be renamed as `latestTest` and `B` to be renamed as `previousTest` or similar - although `B` is actually the `latest` and `A` is the `previous` 🤯